packages/caliper-core/test/worker/rate-control/fixedFeedbackRate.js

```diff
@@ -144,5 +144,29 @@ describe('fixedFeedbackRate controller implementation', () => {
             sinon.assert.calledOnce(sleepStub);
             sinon.assert.calledWith(sleepStub, 20000);
         });
+
+        it('should sleep when no successful transactions have occurred', async () => {
+            txnStats.stats.txCounters.totalSubmitted = 5;
+            txnStats.stats.txCounters.totalFinished = 2;
+            txnStats.stats.txCounters.totalSuccessful = 0;
+            controller.generalSleepTime = 1;
+
+            await controller.applyRateControl();
+
+            sinon.assert.calledOnce(sleepStub);
+            sinon.assert.calledWith(sleepStub, 1 * controller.backOffTime);
+        });
+
+        it('should sleep when unfinished transactions are above the threshold', async () => {
```

Comment:
Unfortunately this doesn't tell me exactly what is being tested here. I know it's about unfinished transactions but have no idea what you mean by the "threshold". So as someone who wants to look at the tests to understand the behaviour using terms that are not defined as part of the configuration of the rate controller is only going to confuse. 

But this also brings me to the code around this area which is lacking. I can only assume you weren't trying to test the following code based on the test
```
        let diff = (this.generalSleepTime * currentSubmitted - ((Date.now() - this.totalSleepTime) - this.stats.getRoundStartTime()));
        if (diff > 5) {
            await util.sleep(diff);
            return;
        }
   ```
but if you were, then the issue with the code is that the variable `diff` is no meaning whatsoever as to the value it represents. To be honest I would have to spend time thinking about it myself to understand what it actually represents and why the value 5 was chosen. This is not technically a bug, but is certainly poor programming as code needs to convey as much information as possible and where that cannot be done then you use comments. 

The point I am making here is that this description doesn't explain the test to me because of the use of the word threshold.